war/src/main/webapp/scripts/hudson-behavior.js

```diff
@@ -198,6 +198,13 @@ var FormChecker = {
     // won't work.
     inProgress : 0,
 
+    // defines the maximum number of parallel checks to be run
+    // plugins that intend to to change this should be aware that when using http1.1 the
+    // browsers will usually throttle the number of connections and having a higher value
+    // can even have a negative impact. But with http2 enabled, this can be a great
+    // performance improvement
```

Comment:
Isn't this something core can take care of? Why would every plugin need to look at the HTTP version?

Reply:
not sure how easily and reliably it can be checked within javascript if http2 is used or not.

Reply:
this can be used apparently:
https://stackoverflow.com/a/46115491/4951015

Reply:
Still likely better handled in core if we can I think then it fixes it for everyone

Reply:
I see you have, will check it later

Reply:
I checked in safari and it works there too

Reply:
the comment above this seems outdated now, as this isn't intended for plugins to change anymore from what I can tell? (given that it is unconditionally overridden on h2)

Reply:
I think with a value of 30 there would be no reason for matrix-auth or role-strategy plugins to overwrite that themselves.
But they would still be able to do so if they want.

Reply:
> performance.getEntries()[0].nextHopProtocol worked with chrome but didn't work with firefox.

If I do `performance.getEntries()` in console of this PR, I get ~100 items, some of them `h2`, some of them not (both Chrome and FF). Could it be that in your case FF and Chrome just had a different resource at position 0?